dpgen2/exploration/report/report_adaptive_lower.py

```diff
@@ -341,6 +387,84 @@ def _get_candidates(
             ret = self.candi_picked
         return ret
 
+    def _get_candidates_inv_pop_f(
+        self,
+        max_nframes: Optional[int] = None,
+    ) -> List[Tuple[int, int]]:
+        """
+        Get candidates. If number of candidates is larger than `max_nframes`,
+        then randomly pick `max_nframes` frames from the candidates.
+        The probability of chose a frame is propotional to the inverse
+        population in force model deviation statistics.
+
+        Parameters
+        ----------
+        max_nframes
+            The maximal number of frames of candidates.
+
+        Returns
+        -------
+        cand_frames   List[Tuple[int,int]]
+            Candidate frames. A list of tuples: [(traj_idx, frame_idx), ...]
+        """
+        self.candi_picked = [(ii[0], ii[1]) for ii in self.candi]
+        if max_nframes is not None and max_nframes < len(self.candi_picked):
+            prob = self._choice_prob_inv_pop_f(self.candi_picked)
+            ret = random.choices(
+                self.candi_picked,
+                weights=prob,
+                k=max_nframes,
+            )
+        else:
+            ret = self.candi_picked
+        return ret
+
+    def _choice_prob_inv_pop_f(
+        self,
+        candi: List,
+    ):
+        """Compute the probability of candi frames according to the inverse
+        population in the model deviation statistics.
+
+        Parameters
+        ----------
+        candi   List[Tuple[int,int]]
+            Candidate frames. A list of tuples: [(traj_idx, frame_idx), ...]
+
+        Returns
+        -------
+        prob    List[float]
+            The probability of each candidate frame.
+
+        """
+        histo = np.zeros(self.nhist)
```

Comment:
It seems to be int? By default dtype is float64.

```suggestion
        histo = np.zeros(self.nhist, dtype=int)
```